File: codegen/results/v002/ollama/qwen3.5_4b_nothink/20260412_065224/typescript/004_json_filter/solution.ts
import * as fs from 'fs';

// Read input file
const inputFile = 'input/data.json';
let data: any[];
try {
  const fileContent = fs.readFileSync(inputFile, 'utf-8');
  data = JSON.parse(fileContent);
} catch (error) {
  console.error(`Error reading ${inputFile}: ${error}`);
  process.exit(1);
}

// Filter and transform
const filtered = data.filter(record => {
  return (
    record.active === true &&
    record.age >= 30
  );
});

// Sort by name ascending
const sorted = filtered.sort((a, b) => {
  return a.name.localeCompare(b.name);
});

// Output to stdout
console.log(JSON.stringify(sorted, null, 2));

File: codegen/results/v002/ollama/qwen3.5_4b_nothink/20260412_065224/go/004_json_filter/solution.go
package main

import (
	"encoding/json"
	"fmt"
	"io/ioutil"
	"sort"
)

type Person struct {
	Name  string `json:"name"`
	Age   int    `json:"age"`
	Active bool  `json:"active"`
	Score float64 `json:"score"`
}

func main() {
	// Read input data
	inputJSON, err := ioutil.ReadFile("input/data.json")
	if err != nil {
		fmt.Fprintln(err.Error)
		return
	}

	var people []Person
	if err := json.Unmarshal(inputJSON, &people); err != nil {
		fmt.Fprintln(err.Error)
		return
	}

	// Filter people: active is true and age >= 30
	filtered := []Person{}
	for _, p := range people {
		if p.Active && p.Age >= 30 {
			filtered = append(filtered, p)
		}
	}

	// Sort by name ascending
	sort.Slice(filtered, func(i, j int) bool {
		return filtered[i].Name < filtered[j].Name
	})

	// Output as JSON
	outputJSON, err := json.MarshalIndent(filtered, "", "  ")
	if err != nil {
		fmt.Fprintln(err.Error)
		return
	}

	fmt.Println(string(outputJSON))
}